### DISCUSSION TITLE
Search & Lens as a "Block"

### DISCUSSION THREAD

**Discussion by SRyanPlanitar** [GitbookIO/community#195] [Feature Requests]
### What's your feature/idea?

I'd love the Search/Lens function on the main page (right in the customer's face). This would be extremely valuable to the usage of the Search and AI Lens function, as right now; it's not natural for customers to look to the top right for search functionality. People are used to a "Google-like" search functionality where the search bar is in the middle main page of the site.

We would appreciate the option to insert this as a block!
![Search bar request](https://user-images.githubusercontent.com/127128944/236260899-3a4009ca-7f7d-4d89-8740-b8e0c974bcd3.jpg)

### What's your desired outcome?

To have the search bar available in multiple places (primarily the landing page)

### What's the impact of this for you?

- [ ] Critical
- [X] High
- [ ] Medium
- [ ] Low

**Comment by addisonschultz** [comment]
Hey @SRyanPlanitar! Great idea - I think creating something like this is already possible! We've been working on GitBook's integration platform, which allows you to create custom blocks and use our REST API to interact with GitBook items/pages/spaces programatically. We even have methods for using Lens!

You can view the docs here! https://developer.gitbook.com/

Specifically, you might be interested in these sections: 

Custom Blocks: https://developer.gitbook.com/integrations/integrations
Lens API: https://developer.gitbook.com/gitbook-api/reference/search#post-ai-ask-question-on-all-content-accessible-by-the-authenticated-user

**Reply by SRyanPlanitar** [reply]
Thank you for this!

**Reply by addisonschultz** [reply]
No problem!

Let me know if you have any questions that come up from here.
